cpp/unreal_plugins/WheeledVehicle/Source/WheeledVehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,177 @@
+
+    // Setup IMU sensor
+    FVector imu_location(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_X"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Y"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Z"));
+
+    FRotator imu_orientation(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.PITCH"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.YAW"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROLL"));
+
+    imu_component_ = CreateDefaultSubobject<UBoxComponent>(TEXT("AVehiclePawn::imu_component_"));
+    ASSERT(imu_component_);
+
+    imu_component_->SetRelativeLocationAndRotation(imu_location, imu_orientation);
+    imu_component_->SetupAttachment(skeletal_mesh_component_);
+
+    // Setup Sonar sensor
+    FVector sonar_location(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_X"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_Y"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.POSITION_Z"));
+
+    FRotator sonar_orientation(
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.PITCH"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.YAW"),
+        Config::get<float>("WHEELED_VEHICLE.VEHICLE_PAWN.SONAR_COMPONENT.ROLL"));
+
+    sonar_component_ = CreateDefaultSubobject<UBoxComponent>(TEXT("AVehiclePawn::sonar_component_"));
+    ASSERT(sonar_component_);
+
+    sonar_component_->SetRelativeLocationAndRotation(sonar_location, sonar_orientation);
+    sonar_component_->SetupAttachment(skeletal_mesh_component_);
+}
+
+AVehiclePawn::~AVehiclePawn()
+{
+    SP_LOG_CURRENT_FUNCTION();
+}
+
+void AVehiclePawn::SetupPlayerInputComponent(UInputComponent* input_component)
+{
+    SP_ASSERT(input_component);
+    APawn::SetupPlayerInputComponent(input_component);
+
+    input_component->BindAxis("MoveForward", this, &AVehiclePawn::moveForward);
+    input_component->BindAxis("MoveRight", this, &AVehiclePawn::moveRight);
+}
+
+void AVehiclePawn::moveForward(float forward)
+{
+    float torque = forward * 10.0;
+    vehicle_movement_component_->SetDriveTorque(torque, 0);
+    vehicle_movement_component_->SetDriveTorque(torque, 1);
+    vehicle_movement_component_->SetDriveTorque(torque, 2);
+    vehicle_movement_component_->SetDriveTorque(torque, 3);
+}
+
+void AVehiclePawn::moveRight(float right)
+{
+    float torque = right * 10.0;
+    vehicle_movement_component_->SetDriveTorque(torque, 0);
+    vehicle_movement_component_->SetDriveTorque(-1.0 * torque, 1);
+    vehicle_movement_component_->SetDriveTorque(torque, 2);
+    vehicle_movement_component_->SetDriveTorque(-1.0 * torque, 3);
+}
+
+// Apply the drive torque in[N.m] to the vehicle wheels.The applied driveTorque persists until the
```

Comment:
You don't need any of these functions because `vehicle_movement_component_` is public. You can do all of this directly at the call site.

Reply:
Agreed, updated this